tests/python/gaia-ui-tests/gaiatest/gaia_test.py

```diff
@@ -739,26 +744,41 @@ def is_screen_enabled(self):
         return ret
 
     def touch_home_button(self):
+        from gaiatest.apps.lockscreen.app import LockScreen
+        from gaiatest.apps.homescreen.app import Homescreen
+
         apps = GaiaApps(self.marionette)
-        if apps.displayed_app.name.lower() != 'homescreen':
-            # touching home button will return to homescreen
-            self._dispatch_home_button_event()
-            Wait(self.marionette).until(
-                lambda m: apps.displayed_app.name.lower() == 'homescreen')
-            apps.switch_to_displayed_app()
+        apps.switch_to_displayed_app()
+
+        if apps.displayed_app.origin == Homescreen.origin:
```

Comment:
Note: I'll update the PR, once [bug 1188926](https://bugzilla.mozilla.org/show_bug.cgi?id=1188926) lands